src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -653,9 +669,39 @@ public CompilationInfo(Compilation compilation, bool hasSuccessfullyLoaded)
                         }
                     }
 
+                    // Now we run generators; if we don't have a generator driver at all, we must try create one
+                    if (generatorDriver.GeneratorDriver == null)
+                    {
+                        var generators = this.ProjectState.AnalyzerReferences.SelectMany(a => a.GetGenerators()).ToImmutableArray();
+                        var additionalTexts = this.ProjectState.AdditionalDocumentStates.Values.SelectAsArray(a => (AdditionalText)new AdditionalTextWithState(a));
+                        var compilationFactory = this.ProjectState.LanguageServices.GetRequiredService<ICompilationFactoryService>();
+
+                        generatorDriver = new TrackedGeneratorDriver(
+                            compilationFactory.CreateGeneratorDriver(
+                                this.ProjectState.ParseOptions!,
+                                generators,
+                                additionalTexts));
```

Comment:
This will drop the value of `generatorDriver.NeedsFullGeneration` if it was `false`. Is there ever a case where that could happen?

Reply:
The NeedsFullGeneration flag is to track whether or not the GeneratorDriver.TryApplyEdits would be useful; that is only useful if you had an existing generator around, so once we're creating a new one we need a full generation no matter what.